src/downloadAzureProject/setupProjectFolder.ts

```diff
@@ -0,0 +1,101 @@
+
+        await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: localize('settingUpFunctionAppLocalProjInfoMessage', `Setting up project for function app '${functionAppName}' with language '${language}'.`) }, async () => {
+            if (downloadAppContent) {
+                // NOTE: We don't want to download app content for compiled languages.
+                const slotTreeItem: SlotTreeItemBase | undefined = await ext.tree.findTreeItem(resourceId, { ...context, loadAll: true });
+                const hostKeys: WebSiteManagementModels.HostKeys | undefined = await slotTreeItem?.client.listHostKeys();
+                const defaultHostName: string | undefined = slotTreeItem?.client.defaultHostName;
+
+                if (!!hostKeys && hostKeys.masterKey && defaultHostName) {
+                    const requestOptions: RequestPrepareOptions = {
+                        method: 'GET',
+                        url: `https://${defaultHostName}/admin/functions/download?includeCsproj=true&includeAppSettings=true`,
+                        headers: { 'x-functions-key':  hostKeys.masterKey }
+                    };
+                    await requestUtils.downloadFile(requestOptions, downloadFilePath);
+                }
+            }
+
+            const projectFilePathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, `${functionAppName}`);
+            const projectFilePath: string = projectFilePathUri.fsPath;
+            const devContainerFolderPathUri: vscode.Uri = vscode.Uri.joinPath(projectFilePathUri, '.devcontainer');
+            if (downloadAppContent) {
+                // tslint:disable-next-line: no-unsafe-any
+                await extract(downloadFilePath, { dir: projectFilePath });
+            }
+            await requestUtils.downloadFile(
+                `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/devcontainer.json`,
+                vscode.Uri.joinPath(devContainerFolderPathUri, 'devcontainer.json').fsPath
+            );
+            await requestUtils.downloadFile(
+                `https://raw.githubusercontent.com/microsoft/vscode-dev-containers/master/containers/${devContainerName}/.devcontainer/Dockerfile`,
+                vscode.Uri.joinPath(devContainerFolderPathUri, 'Dockerfile').fsPath
+            );
+            await initProjectForVSCode(context, projectFilePath, getProjectLanguageForLanguage(language));
+            vscode.window.showInformationMessage(localize('restartingVsCodeInfoMessage', 'Restarting VS Code with your function app project'));
+            vscode.commands.executeCommand('vscode.openFolder', vscode.Uri.file(projectFilePath), true);
+        });
+    } catch (err) {
+        vscode.window.showErrorMessage(localize('failedLocalProjSetupErrorMessage', 'Failed to set up your local project. Please try again.'));
```

Comment:
I feel like this generic message isn't very helpful, so I'd prefer to get rid of the catch block and let the original error bubble up. If you want to keep the generic message, please do something [like this](https://github.com/microsoft/vscode-azurefunctions/blob/main/src/templates/script/ScriptBundleTemplateProvider.ts#L72) where you include the original error in the message and use `throw` instead of `showErrorMessage`

Reply:
I am keeping the try catch and going with the second option of throwing the error